### PR TITLE
termpaint: init at 0.3.0

### PR DESCRIPTION
**Pull request by laalsaas**
## Description of changes

I did this PR together with @textshell and @istoph at FrosCon, they'll maintain this packages. They'll approve this PR in order to assure this.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8940,6 +8940,12 @@
     githubId = 13622947;
     keys = [ { fingerprint = "1412 816B A9FA F62F D051 1975 D3E1 B013 B463 1293"; } ];
   };
+  istoph = {
+    email = "chr@istoph.de";
+    name = "Christoph Hüffelmann";
+    github = "istoph";
+    githubID = 114227790;
```
Comment (textshell) [inline maintainers/maintainer-list.nix:8947]:
```suggestion
    githubId = 114227790;
```

Obligatory typo right before finishing.

**Review by textshell — APPROVED** [review]
Thanks.
I think fixing the one typo might get the CI green.

**Review by istoph — APPROVED** [review]

**Review by textshell — APPROVED** [review]

**Review by istoph — APPROVED** [review]

File: pkgs/by-name/te/termpaint/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "-Dttyrescue-fexec-blob=false"
+    "-Dtools-path=libexec/"
+    "-Dttyrescue-path=libexec/"
+    "-Dttyrescue-install=true"
+  ];
+
+  doCheck = true;
+
+  meta = {
+    description = "low level terminal interface library";
```
Comment (Aleksanaa) [inline pkgs/by-name/te/termpaint/package.nix:42]:
```suggestion
    description = "Low level terminal interface library";
    homepage = "https://github.com/termpaint/termpaint";
```

File: pkgs/by-name/te/termpaint/package.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    rev = final.version;
+    hash = "sha256-AsbUJjz51pedmemI0racMgWRzpbIeNJrK/walFUniR4=";
+  };
+
+  patches = [ ./0001-meson.build-use-prefix.patch ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    cmake
```
Comment (Aleksanaa) [inline pkgs/by-name/te/termpaint/package.nix:27]: If you remove it, will build fail?

**Review by textshell — APPROVED** [review]

**Review by istoph — APPROVED** [review]

File: pkgs/by-name/te/termpaint/package.nix (hunk 2 above)
Comment (Aleksanaa) [reply pkgs/by-name/te/termpaint/package.nix]: This is not resolved

**Pull request merged**